src/operations/find_and_modify.ts

```diff
@@ -117,4 +116,78 @@ class FindAndModifyOperation extends OperationBase {
   }
 }
 
-export = FindAndModifyOperation;
+/**
+ * Execute a command
+ *
+ * @function
+ * @param {Db} db The Db instance on which to execute the command.
+ * @param {object} command The command hash
+ * @param {object} [options] Optional settings. See Db.prototype.command for a list of options.
+ * @param {Db~resultCallback} [callback] The command result callback
+ */
+function executeCommand(db: any, command: object, options?: any, callback?: Function) {
+  // Did the user destroy the topology
+  if (db.serverConfig && db.serverConfig.isDestroyed()) {
+    return callback!(new MongoError('topology was destroyed'));
+  }
+
+  // Convert the readPreference if its not a write
+  options.readPreference = ReadPreference.resolve(db, options);
+
+  // Execute command
+  db.s.topology.command(
+    db.s.namespace.withCollection('$cmd'),
+    command,
+    options,
+    (err?: any, result?: any) => {
+      if (err) return handleCallback(callback!, err);
+      if (options.full) return handleCallback(callback!, null, result);
+      handleCallback(callback!, null, result.result);
+    }
+  );
+}
+
+class FindOneAndDeleteOperation extends FindAndModifyOperation {
+  constructor(collection: any, filter: any, options: any) {
+    // Final options
+    const finalOptions = Object.assign({}, options);
+    finalOptions.fields = options.projection;
+    finalOptions.remove = true;
+
+    super(collection, filter, finalOptions.sort, null, finalOptions);
+  }
+}
+
+class FindOneAndReplaceOperation extends FindAndModifyOperation {
+  constructor(collection: any, filter: any, replacement: any, options: any) {
+    // Final options
+    const finalOptions = Object.assign({}, options);
+    finalOptions.fields = options.projection;
+    finalOptions.update = true;
+    finalOptions.new = options.returnOriginal !== void 0 ? !options.returnOriginal : false;
+    finalOptions.upsert = options.upsert !== void 0 ? !!options.upsert : false;
```

Comment:
would it be equivalent here to use `undefined` as opposed to `void 0`? I think `undefined` is clearer

Reply:
Ok yea my mistake for not considering "refactor:" 😅 but thanks anyway for addressing my comments with ticket links, its good to connect it all